### DISCUSSION TITLE
GLHKViewからHKViewにしたところプレビュー画面が回転しない

### DISCUSSION THREAD

**Discussion by KAZUMA87** [HaishinKit/HaishinKit.swift#1054] [Q&A] [answered]
GLHKViewを利用していたが、バージョン1.2.7にしたところサポートしていないようだったので、HKViewへと変更を行った。
プレビュー表示も配信も問題無く行えたのだが、プレビュー表示だけ回転方向がおかしい状態になってしまっている。
基本的にランドスケープモードの内側カメラ利用をするようにしてあるのだが、プレビュー表示で90度右に回転してしまっている。
プレビューに対して、orientationを設定し、.landscapeRightなどを指定したが、特に変化が無い。
配信された画面では問題の無い表示がされていて、プレビューだけ回転がおかしい状態になっている。

何か解決策はありますでしょうか？

**Comment by shogo4405** [comment, marked as answer]
- `GLHKView`の代替のビューということであれば、`MTHKView`(Metal)も用意しております。
- `HKView`であれば、起動時に次のように値をセットすると期待通り動く可能性があります。
  - HaishinKit.swift/blob/main/Examples/iOS/LiveViewController.swift#L33-L35

**Reply by KAZUMA87** [reply]
ご返答ありがとうございます。
Metalの方が変更が少なく済みそうでしたので、MTHKViewに切り替えたところあっさり問題解決致しました。
最高のライブラリをありがとうございます。
